scripts/cocoapods/utils.rb

```diff
@@ -163,4 +163,26 @@ def self.create_xcode_env_if_missing
 
         system("echo 'export NODE_BINARY=$(command -v node)' > #{file_path}")
     end
+
+    # It examines the target_definition property and sets the appropriate value for
+    # ENV['USE_FRAMEWORKS'] variable.
+    #
+    # - parameter target_definition: The current target definition
+    def self.detect_use_frameworks(target_definition)
+        if ENV['USE_FRAMEWORKS'] != nil
+            return
+        end
+
+        framework_build_type = target_definition.build_type.to_s
+
+        Pod::UI.puts("Framework build type is #{framework_build_type}")
+
+        if framework_build_type === "static framework"
```

Comment:
would be great for pasting the code reference, e.g. https://github.com/CocoaPods/Core/blob/c6ac388ee43f0782fdb8e32386f41f96dbe29082/lib/cocoapods-core/build_type.rb#L104-L106

Reply:
I'll add all the references!